code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -33,11 +33,15 @@
 	var/material_drop_amount = 2
 	var/delivery_icon = "deliverycloset" //which icon to use when packagewrapped. null to be unwrappable.
 	var/anchorable = TRUE
-
+	var/obj/item/electronics/airlock/lockerelectronics //Installed electronics
+	var/doinglockstuff = FALSE //Someone is doing some stuff with the lock here, better not proceed further
 
 /obj/structure/closet/Initialize(mapload)
 	if(mapload && !opened)		// if closed, any item at the crate's loc is put in the contents
 		addtimer(CALLBACK(src, .proc/take_contents), 0)
+	if(secure)

```

Comment:
@FlattestGuitar Would you rather me do initialising the electronics this way, or would you rather me make a bunch of subtypes of `obj/item/electronics/airlock` and setting `lockerelectronics` to those paths?

Reply:
The current implementation is fine.